dask_jobqueue/deploy/cluster_manager.py

```diff
@@ -122,6 +131,7 @@ def adapt(self, minimum_cores=None, maximum_cores=None,
                 kwargs['maximum'] = self._get_nb_workers_from_memory(maximum_memory)
         self._adaptive_options.update(kwargs)
         self._adaptive = Adaptive(self.scheduler, self, **self._adaptive_options)
+        self.wait_until_n_workers(wait_until_n)
```

Comment:
Adapt can be called within the event loop, so we can't have a `while ...: time.sleep()` call in there, otherwise things will halt/block.